lib/response_bank/response_cache_handler.rb

```diff
@@ -145,13 +145,16 @@ def serve_from_cache(cache_key_hash, match_entity_tag = "*", cache_age_tolerance
         # regen
         @headers.merge!(headers)
 
-        unless @headers['Content-Encoding']
-          ResponseBank.log("Cache hit, but missing content-encoding in the cache value headers")
-          return
+        if @headers['Content-Encoding'].nil?
+          unless body.nil? || body.empty?
+            ResponseBank.log("Cache hit, but missing content-encoding in the cache value headers")
+          else
+            ResponseBank.log("Cache hit, but empty body")
+          end
         end
 
         # if a cache key hit and client doesn't match encoding, return the raw body
-        if !@env['HTTP_ACCEPT_ENCODING'].to_s.include?(@headers['Content-Encoding'])
+        if !body.nil? && !body.empty? && !@headers['Content-Encoding'].nil? && !@env['HTTP_ACCEPT_ENCODING'].to_s.include?(@headers['Content-Encoding'])
           ResponseBank.log("uncompressing payload for client as client doesn't require encoding")
           body = ResponseBank.decompress(body, @headers['Content-Encoding'])
```

Comment:
the problem is that 304 and 404s do not have a content-encoding header but are cached. so the `.include?` fails because the content-encoding is missing.